src/CircleMenu/CircleMenu.jsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import CircleMenuItem from './CircleMenuItem/CircleMenuItem.jsx';
+
+import './CircleMenu.less';
+
+/**
+ * The CircleMenu.
+ *
+ * @class CircleMenu
+ * @extends React.Component
+ */
+export class CircleMenu extends React.Component {
+
+    _ref = null;
+
+  static propTypes = {
+    /**
+     * The duration of the animation in milliseconds. Pass 0 to avoid animation.
+     * Default is 300.
+     *
+     * @type {Number}
+     */
+    animationDuration: PropTypes.number,
+    /**
+     * The diameter of the CircleMenu. Default is 100.
+     *
+     * @type {Number}
+     */
+    diameter: PropTypes.number,
+    /**
+     * The children of the CircleMenu. Most common are buttons.
+     *
+     * @type {Node}
+     */
+    children: PropTypes.node.isRequired,
+    /**
+     * An Array containing the x and y coordinates of the CircleMenus Center.
+     * @type {Number[]}
+     */
+    position: PropTypes.arrayOf(PropTypes.number).isRequired
+  };
+
+  static defaultProps = {
+    animationDuration: 300,
+    diameter: 100
+  };
+
+  /**
+   * Create the CircleMenu.
+   *
+   * @param {Object} props The initial props.
+   * @constructs CircleMenu
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: props.children
+    };
+  }
+
+  /**
+   * A react lifecylce method called when the component did mount.
+   * It calls the applyTransformation method right after mounting.
+   */
+  componentDidMount() {
+    setTimeout(this.applyTransformation.bind(this), 1);
```

Comment:
Is this the suggested way of doing it ASAP but non-blocking?

Reply:
This is by far the most easiest way to do it yes. I don't like it too but the alterantive is using a package like [react-transition-group](https://github.com/reactjs/react-transition-group) which is really uncomfortable and inflated.